LibreNMS/OS/Shared/Cisco.php

```diff
@@ -641,6 +641,10 @@ public function discoverTransceivers(): Collection
             $data = $this->getDevice()->entityPhysical()->whereIn('entPhysicalContainedIn', $dbSfpCages->keys())->get()->map(function ($ent) use ($dbSfpCages) {
                 if (empty($ent->ifIndex) && $dbSfpCages->has($ent->entPhysicalContainedIn)) {
                     $ent->ifIndex = $dbSfpCages->get($ent->entPhysicalContainedIn);
+                    if (empty($ent->ifIndex)) {
+                        // Lets try to find the 1st subentity with an ifIndex below this one and use it. A few ISR on IOSXE at least are behaving like this.
+                        $ent->ifIndex = $this->getDevice()->entityPhysical()->whereIn('entPhysicalContainedIn', [$ent->entPhysicalIndex])->whereNotNull('ifIndex')->first()->ifIndex;
```

Comment:
It would be better to do one query to make a map then a query for each transceiver.

Reply:
Yep, makes sense. Trying to check first if I have regressions or not (and fight with my test-data generation currently). But will do before removing WIP

Reply:
In fact, computing all the values when we may only need a couple (you normally have a bunch of ENTITY entries compared to the number of SFPs, even more if you consider this code is so far only necessary on ASRs and ISR which usually have only a small number of SFPs) is probably not the best idea. And I don't see an easy way to change the request to achieve it anyway. 
Still looking at that.